### PR TITLE
CHANGELOG Completion for VM v5 Release

### PR DESCRIPTION
**Pull request by holgerd77**
Completion of the CHANGELOG files for the various libaries in preparation for the VM v5 release.

Dates on latest included merges are added to the commit messages, so some last PRs can be added upon the final release PRs (or - better - along eventually following feature PRs).

Updates here will be kept in sync with the TODO list from #907 upon pushes.

### CODE REVIEW

File: packages/blockchain/CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,24 @@ The deprecated `validate` option has been removed, please use `valdiateBlock` an
...
+  PR [#748](https://github.com/ethereumjs/ethereumjs-vm/pull/748)
+
+**Bug Fixes**
+
+- Fixed blockchain hanging forever in case code throws between a semaphore `lock`/`unlock`,
+  Issue [#877](https://github.com/ethereumjs/ethereumjs-vm/issues/877)
+
+## 4.0.4 - 2020-07-27
+
+This release replaces the tiled (`~`) dependency from `ethereumjs-util` for a caret (`^`) one, meaning that any update to `ethereumjs-util` v6 will also be available for this library.
```
Comment (cgewecke) [inline packages/blockchain/CHANGELOG.md:80]:
```suggestion
This release replaces the tilde (`~`) dependency from `ethereumjs-util` for a caret (`^`) one, meaning that any update to `ethereumjs-util` v6 will also be available for this library.
```

File: packages/common/CHANGELOG.md [hunk 2]
```diff
@@ -19,13 +31,76 @@ import Common from '@ethereumjs/common'
...
+```
+
+The following EIPs are initially supported within this release:
+
+- [EIP-2537](https://eips.ethereum.org/EIPS/eip-2315) BLS Precompiles
+- [EIP-2315](https://eips.ethereum.org/EIPS/eip-2315) EVM Subroutines (PR [#876](https://github.com/ethereumjs/ethereumjs-vm/pull/876))
+
+EIPs provided are then activated and parameters requested with `Common.param()` being present in these EIPs take precedence over the setting from the latest hardfork.
+
+There are two new utility functions which dedicatedly return the hardfork respectively EIP values:
```
Comment (cgewecke) [inline packages/common/CHANGELOG.md:49]:
```suggestion
There are two new utility functions which return hardfork and EIP values respectively:
```

File: packages/common/CHANGELOG.md [hunk 3]
```diff
@@ -19,13 +31,76 @@ import Common from '@ethereumjs/common'
...
+
+For setting and requesting active EIPs there is `Common.setEIPs()` and `Common.eips()` added to the mix.
 
-A new function `setHardforkByBlockNumber()` has been added, PR [#863](https://github.com/ethereumjs/ethereumjs-vm/pull/863)
+Along there is a new EIP-based hardfork file format which allows to just reference EIPs (pointing to the EIP files which then contain the parameters) instead of containing parameters directly, see PR [#876](https://github.com/ethereumjs/ethereumjs-vm/pull/876). This is in preparation for an upcoming `Yolo v2` testnet integration.
 
-### Default hardfork
+Side note: with this new structural setup it gets now possible for all EIPs still implicitly contained within the hardfork files to be extracted as an EIP parameter set within its own dedicated EIP file (which can then be activated via the `eip` parameter on initialization) without loosing on functionality. If you have a need there feel free to open a PR!
```
Comment (cgewecke) [inline packages/common/CHANGELOG.md:60]:
```suggestion
Side note: with this new structural setup it's now possible for all EIPs still implicitly contained within the hardfork files to be extracted as an EIP parameter set within its own dedicated EIP file (which can then be activated via the `eip` parameter on initialization) without losing any functionality. If you have a need there feel free to open a PR!
```

File: packages/common/CHANGELOG.md [hunk 4]
```diff
@@ -19,13 +31,76 @@ import Common from '@ethereumjs/common'
...
+
+### Eth/64 Forkhash Support
+
+There is a new `Common.forkHash()` method returning pre-calculated Forkhash values or alternatively use the internal `Common._calcForkHash()` implementation to calculate a forkhash on the fly.
+
+Forkhashes are used to uniquely identify a set of hardforks passed to be able to better differentiate between different dedicated chains. This is used for the `Eth/64` devp2p protocol update and specificed in [EIP-2124](https://eips.ethereum.org/EIPS/eip-2124) to help improve the devp2p networking stack.
+
+### New Block/Hardfork related Utility Functions
+
+The following block respectively hardfork related utility functions have been added along PR [#863](https://github.com/ethereumjs/ethereumjs-vm/pull/863) and PR [#805](https://github.com/ethereumjs/ethereumjs-vm/pull/805):
```
Comment (cgewecke) [inline packages/common/CHANGELOG.md:75]:
```suggestion
The following block and hardfork related utility functions have been added with PRs [#863](https://github.com/ethereumjs/ethereumjs-vm/pull/863) and [#805](https://github.com/ethereumjs/ethereumjs-vm/pull/805) respectively:
```

File: packages/vm/CHANGELOG.md [hunk 5]
```diff
@@ -98,30 +98,69 @@ and can be passed in on instantiation have been updated to new major versions.
...
 All these libraries are now written in `TypeScript` and use promises instead of
 callbacks for accessing their APIs.
 
-### Other Additions/Changes
+### New StateManager Interface
+
+There is now a new `TypeScript` interface for the `StateManager`, see
+PR [#763](https://github.com/ethereumjs/ethereumjs-vm/pull/763). If you are
+using a custom `StateManager` you can use this interface to get better
+assurance that you are using a `StateManager` going conform with the current
```
Comment (cgewecke) [inline packages/vm/CHANGELOG.md:113]:
```suggestion
assurance that you are using a `StateManager` which conforms with the current
```

File: packages/vm/CHANGELOG.md [hunk 6]
```diff
@@ -98,30 +98,69 @@ and can be passed in on instantiation have been updated to new major versions.
...
+`StateManager` API and therefore running in the VM without problems.
```
Comment (cgewecke) [inline packages/vm/CHANGELOG.md:114]:
```suggestion
`StateManager` API and will run in the VM without problems.
```

**Review by cgewecke — APPROVED** [review]
This all makes sense & looks great to me  🎉  


(Just left a couple re-wording suggestions above...)

File: packages/common/CHANGELOG.md (hunk 2 above)
Comment (holgerd77) [reply packages/common/CHANGELOG.md:49]: Will mark as resolved along fixing locally and push once finished on all the change requests.

**Comment by holgerd77** [comment]
Rebased this, integrated the suggested re-writes and did a first write on the Bock library refactoring changes. This is still WIP (the Block CHANGELOG is more extensive than I first thought).

**Comment by holgerd77** [comment]
Ok, this is now ready for review! 😄

**Review by ryanio — APPROVED** [review]
whew, big changelogs, looks really good!

**Pull request merged**